docker/api/Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM node:16-buster AS builder
+# Install doppler CLI
+RUN (curl -Ls --tlsv1.2 --proto "=https" --retry 3 https://cli.doppler.com/install.sh) | sh -s -- --verify-signature
+# node images create a non-root user that we can use
+USER node
+WORKDIR /home/node/build
+COPY --chown=node:node . .
+# Pass `DOPPLER_TOKEN` at build time to create an encrypted snapshot for high-availability
+ARG DOPPLER_TOKEN
+RUN \
+  doppler secrets download doppler.encrypted.json &&\
+  npm ci --no-progress --ignore-scripts &&\
+  doppler run --fallback=doppler.encrypted.json --command="npm run create:config" &&\
+  doppler run --fallback=doppler.encrypted.json --command="npm run build:curriculum" &&\
+  doppler run --fallback=doppler.encrypted.json --command="npm run build:server"
+
+FROM node:16-alpine as depends
+USER node
+WORKDIR /home/node/depends
+COPY --chown=node:node . .
+RUN npm ci --production --workspace=api-server --include-workspace-root --no-progress --ignore-scripts
+
+FROM node:16-alpine
+RUN npm i -g pm2@4
+USER node
+WORKDIR /home/node/api
+COPY --from=builder --chown=node:node /home/node/build/api-server/lib/ api-server/lib/
+COPY --from=builder --chown=node:node /home/node/build/utils/ utils/
+COPY --from=builder --chown=node:node /home/node/build/config/ config/
+COPY --from=depends --chown=node:node /home/node/depends/api-server/node_modules/ api-server/node_modules/
+COPY --from=depends --chown=node:node /home/node/depends/node_modules/ node_modules/
```

Comment:
I am not a fan of this, but as is this is required for several things like `debug`, `cross-env`, etc. to work. Unfortunately, they are installed at the top level.

Reply:
Thanks for the feedback. I will go ahead and add that change in this PR.